core/src/spotify_id.rs

```diff
@@ -13,7 +18,7 @@ const BASE62_DIGITS: &'static [u8] = b"0123456789abcdefghijklmnopqrstuvwxyzABCDE
 const BASE16_DIGITS: &'static [u8] = b"0123456789abcdef";
 
 impl SpotifyId {
-    pub fn from_base16(id: &str) -> Result<SpotifyId, SpotifyIdError> {
+    pub fn from_base16(id: &str, podcast: bool) -> Result<SpotifyId, SpotifyIdError> {
```

Comment:
This would be a breaking API change - might be better to pares the URI and do this automatically, from `from_rawURI`

Reply:
I also changed SpotifyId to be an enum now, not a struct.
Would also break API, but I think this is more clean than a big If/Else solution

Reply:
Let me dig up what I did back then..

Reply:
Refactored this breaking change now, thanks for the review